src/libmongoc/src/mongoc/mongoc-async-cmd.c

```diff
@@ -134,20 +136,38 @@ mongoc_async_cmd_run (mongoc_async_cmd_t *acmd)
 }
 
 void
-_mongoc_async_cmd_init_send (mongoc_async_cmd_t *acmd, const char *dbname)
+_mongoc_async_cmd_init_send (const mongoc_opcode_t cmd_opcode, mongoc_async_cmd_t *acmd, const char *dbname)
 {
+   acmd->rpc.header.msg_len = 0; 
+   acmd->rpc.header.request_id = ++acmd->async->request_id; /* in _server_monitor_awaitable_hello_send(), this is postfix++ */
```

Comment:
I think the `postfix++` / `++prefix` is arbitrary. As long as the request_id is unique for each message on a single connection.